Client/Application/AppDelegate.swift

```diff
@@ -283,3 +288,65 @@ extension AppDelegate {
         menuBuilderHelper?.mainMenu(for: builder)
     }
 }
+
+// MARK: - Application Services History Migration
+// A convinient mapping, `Profile.swift` can't depend
+// on `PlacesMigrationConfiguration` directly since
+// the FML is only usable from `Client` at the moment
+extension PlacesMigrationConfiguration {
+    func into() -> HistoryMigrationConfiguration {
+        switch self {
+        case .disabled:
+            return HistoryMigrationConfiguration.disabled
+        case .dryRun:
+            return HistoryMigrationConfiguration.dryRun
+        case .real:
+            return HistoryMigrationConfiguration.real
+        }
+    }
+}
+
+extension AppDelegate {
+    func runAppServicesHistoryMigration() {
+        let placesHistory = FxNimbus.shared.features.placesHistory.value()
+        FxNimbus.shared.features.placesHistory.recordExposure()
+        guard placesHistory.migration != .disabled else {
+            return
+        }
+        let p = self.profile as? BrowserProfile
```

Comment:
nit: could we call it profile value or something more descriptive instead of `let p` 😅

Reply:
Also, is the plan to remove this in future or do we plan on doing more migrations